### PR TITLE
Bump Go version to 1.20

### PR DESCRIPTION
**Pull request by terryyylim**
**What this PR does / why we need it**:

Update Go version to `1.20` prior to integrating inference-logger.

**Which issue(s) this PR fixes**:

`NONE`

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/merlin.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ on:
 env:
   ARTIFACT_RETENTION_DAYS: 7
   DOCKER_BUILDKIT: 1
-  GO_VERSION: 1.18
+  GO_VERSION: '1.20'
```
Comment (krithika369) [inline .github/workflows/merlin.yml:13]: Why the single quotes?

**Review by krithika369 — APPROVED** [review]
LGTM. Thanks!

File: .github/workflows/merlin.yml (hunk 1 above)
Comment (terryyylim) [reply .github/workflows/merlin.yml:13]: It'll be truncated to 1.2 without it and the linting step will fail.

Comment (krithika369) [reply .github/workflows/merlin.yml:13]: Ahh ok.

**Pull request merged**